db/submission_test.go

```diff
@@ -59,3 +59,54 @@ func (this *DBTests) DBTestRemoveSubmission(test *testing.T) {
         }
     }
 }
+
+func (this *DBTests) DBTestFetchAttempts(test *testing.T) {
+    ResetForTesting();
+    defer ResetForTesting();
+
+    assignment := MustGetTestAssignment()
+
+    // User has submissions
+    studentAttempts, err := GetAttempts(assignment, "student@test.com");
+    if err != nil {
+        test.Errorf("Failed to get student attempts when should be a success: %v", err);
+    }
```

Comment:
Put a blank line between the error-check if block and the size-check if block.
I understand the intention of putting them close (they are apart of the same test subcase).
But as a general rule you always* want to put vertical spacing between different if-blocks.
This is because it is very easy for a reader (especially after a few hundred lines) to see the structure and mistakenly see an "else if" where there is the "if" (because the code is structured like that) and think the two code blocks are mutually exclusive.

If we were not writing Go then I would recommend to have the second "if" be an "else if",
but since we are we do not want to mess with the super common error checking pattern of a single if checking if err is not nil.

*99.999% of the time.